src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/custom.py

```diff
@@ -90,11 +114,233 @@ def _upload_package_blob(package_file, url):
+        certificate_format = 'cer'
+    with open(cert_file, "rb") as f:
+        data_bytes = f.read()
+    data = base64.b64encode(data_bytes).decode('utf-8')
+    cert = CertificateAddParameter(thumbprint, thumbprint_algorithm, data,
+                                   certificate_format=certificate_format,
+                                   password=password)
+    try:
+        client.add(cert)
+        return client.get(thumbprint_algorithm, thumbprint)
+    except BatchErrorException as ex:
+        try:
+            message = ex.error.message.value
+            if ex.error.values:
+                for detail in ex.error.values:
+                    message += "\n{}: {}".format(detail.key, detail.value)
+            raise CLIError(message)
+        except AttributeError:
+            raise CLIError(ex)
+    except (ValidationError, ClientRequestError) as ex:
+        raise CLIError(ex)
+
+
+def delete_certificate(client, thumbprint, thumbprint_algorithm, abort=None):
+    try:
+        if abort:
+            client.cancel_deletion(thumbprint_algorithm, thumbprint)
+        else:
+            client.delete(thumbprint_algorithm, thumbprint)
+    except BatchErrorException as ex:
+        try:
+            message = ex.error.message.value
+            if ex.error.values:
+                for detail in ex.error.values:
+                    message += "\n{}: {}".format(detail.key, detail.value)
+            raise CLIError(message)
+        except AttributeError:
+            raise CLIError(ex)
+    except (ValidationError, ClientRequestError) as ex:
+        raise CLIError(ex)
+
+
+@transfer_doc(PoolResizeParameter)
+def resize_pool(client, pool_id, target_dedicated=None,  # pylint:disable=too-many-arguments
+                resize_timeout=None, node_deallocation_option=None,
+                if_match=None, if_none_match=None, if_modified_since=None,
+                if_unmodified_since=None, abort=None):
+    if abort:
+        stop_resize_option = PoolStopResizeOptions(if_match=if_match,
+                                                   if_none_match=if_none_match,
+                                                   if_modified_since=if_modified_since,
+                                                   if_unmodified_since=if_unmodified_since)
+        return client.stop_resize(pool_id, pool_stop_resize_options=stop_resize_option)
+    else:
+        param = PoolResizeParameter(target_dedicated,
+                                    resize_timeout=resize_timeout,
+                                    node_deallocation_option=node_deallocation_option)
+        resize_option = PoolResizeOptions(if_match=if_match,
+                                          if_none_match=if_none_match,
+                                          if_modified_since=if_modified_since,
+                                          if_unmodified_since=if_unmodified_since)
+
+    try:
+        return client.resize(pool_id, param, pool_resize_options=resize_option)
```

Comment:
I question this convention that I've seen in this PR of having an `abort` flag that executes a different operation.
Should consider having these as two separate commands.

Does `client.stop_resize` return the same object structure as `client.resize`?
If not, this is another reason why you may want these as two separate commands.